SingularityService/src/test/java/com/hubspot/singularity/mesos/SingularityStartupTest.java

```diff
@@ -159,4 +167,47 @@ public void testRunOnceDoesntGetRescheduled() {
     // assert that SingularityStartup does not enqueue a SingularityPendingRequest (pendingType=STARTUP) for the RUN_ONCE request
     Assert.assertTrue(requestManager.getPendingRequests().isEmpty());
   }
+
+  @Test
+  public void itFailsIfNumberOfStartupTasksPassThreshold() {
+    int maxPct = 100;
+    Assert.assertTrue("StartUpTaskThresholdPct cannot be above 100%", configuration.getStartUpTaskThresholdPct() <= maxPct);
+
+    // round down to closest multiple of 5 to reduce the number of requests created during this test
```

Comment:
Why don't you just call `configuration.setStartUpTaskThresholdPct()` and set it to something low at the beginning of this test, then set it back to default value at the end. Should save a bunch of extra math, and allow you to set a lower threshold so you don't need to create a ton of requests for the test